### PR TITLE
FR-14999 - Fixed support for vue-router 4 in vue 3

### PR DESCRIPTION
**Pull request by frontegg-david**

### CODE REVIEW

File: packages/vue/src/index.ts [hunk 1]
```diff
@@ -177,10 +177,11 @@ const Frontegg: PluginObject<PluginOptions> | any = {
     }, 10);
 
     const isAuthRoutes = (path: string) => {
+      const pathname = new URL(path, window.location.origin).pathname
```
Comment (doregg) [inline packages/vue/src/index.ts:180]: Previously is was extracted from the currentRoute - how can we make sure it's not a breaking change for other customers? @frontegg-david

File: packages/vue/src/index.ts [hunk 2]
```diff
@@ -177,10 +177,11 @@ const Frontegg: PluginObject<PluginOptions> | any = {
...
       const { routes } = Vue.fronteggAuth;
       return Object.values(routes)
-        .filter(path => path != routes.authenticatedUrl)
-        .includes(path);
+        .filter(route => route != routes.authenticatedUrl)
+        .includes(pathname);
```
Comment (yuvalotem1) [inline packages/vue/src/index.ts:184]: create generic function and use also for `useFronteggAuthGuard` function
```
isAuthRoutes(path, routes){
 const pathname = new URL(path, window.location.origin).pathname
 routes
   .filter(route => route != routes.authenticatedUrl)
   .includes(pathname);
}
```

File: packages/vue/src/auth/mapAuthState.ts [hunk 3]
```diff
@@ -245,14 +245,19 @@ export const useFronteggAuthGuard = (options?: FronteggAuthGuardOptions) => {
...
-      .filter(path => path !== routes.authenticatedUrl)
-      .includes(path);
+      .filter(route => route !== routes.authenticatedUrl)
+      .includes(pathname);
   };
 
   const checkGuard = () => {
-    if (!isAuthRoutes(fronteggAuth.router?.currentRoute.path!) && !authState.isAuthenticated && !authState.isLoading) {
+
+    // @ts-ignore
```
Comment (yuvalotem1) [inline packages/vue/src/auth/mapAuthState.ts:257]: why do we need this ignore?

File: packages/vue/src/index.ts (hunk 1 above)
Comment (yuvalotem1) [reply packages/vue/src/index.ts:180]: This is still extracted from `currentRoute` just for vue router 4 it is taken from `value. fullPath`, why do we not just take the `path` and not `fullpath `and avoid this line? @frontegg-david

File: packages/vue/src/auth/service.ts [hunk 4]
```diff
@@ -152,4 +152,20 @@ export class FronteggAuthService implements FronteggPluginService {
...
+      route = this.router?.currentRoute?.path;
+    }
+    // @ts-ignore vue-router v4
+    if (this.router?.currentRoute?.value?.path) {
+      // @ts-ignore vue-router v4
+      route = this.router?.currentRoute?.value?.fullPath;
+    }
+
+    return route
+  };
```
Comment (yuvalotem1) [inline packages/vue/src/auth/service.ts:170]: maybe we can write typeguard for this that will indicate if is vue 3 or 4 router?

File: packages/vue/src/auth/service.ts [hunk 5]
```diff
@@ -152,4 +152,20 @@ export class FronteggAuthService implements FronteggPluginService {
...
+  getCurrentRoute = () => {
+    let route: string = '/';
+    if (this.router?.currentRoute?.path) {
+      // vue-router v3
+      route = this.router?.currentRoute?.path;
+    }
+    // @ts-ignore vue-router v4
+    if (this.router?.currentRoute?.value?.path) {
+      // @ts-ignore vue-router v4
+      route = this.router?.currentRoute?.value?.fullPath;
```
Comment (yuvalotem1) [inline packages/vue/src/auth/service.ts:166]: why do we check if `this.router?.currentRoute?.value?.path` and then return `this.router?.currentRoute?.value?. fullPath`

File: packages/vue/src/index.ts (hunk 1 above)
Comment (frontegg-david) [reply packages/vue/src/index.ts:180]: this fixes the isAuthRoute bugs.

File: packages/vue/src/auth/mapAuthState.ts [hunk 6]
```diff
@@ -215,8 +215,8 @@ export const useFrontegg = () => {
   const fronteggStore = useFronteggStore() as EnhancedStore;
 
   const loginWithRedirect = () => {
-    // @ts-ignore
-    if (!fronteggAuth.router?.currentRoute.path.startsWith(authState.routes.hostedLoginRedirectUrl)) {
+    const path = fronteggAuth.getCurrentRoute()
```
Comment (doregg) [inline packages/vue/src/auth/mapAuthState.ts:218]:
```suggestion
    const path = fronteggAuth.getCurrentRoute();
```

File: packages/vue/src/auth/mapAuthState.ts [hunk 7]
```diff
@@ -245,14 +245,17 @@ export const useFronteggAuthGuard = (options?: FronteggAuthGuardOptions) => {
   const fronteggStore = useFronteggStore() as EnhancedStore;
 
   const isAuthRoutes = (path: string) => {
+    const pathname = new URL(path, window.location.origin).pathname
```
Comment (doregg) [inline packages/vue/src/auth/mapAuthState.ts:248]:
```suggestion
    const pathname = new URL(path, window.location.origin).pathname;
```

File: packages/vue/src/index.ts (hunk 2 above)
Comment (frontegg-david) [reply packages/vue/src/index.ts]: added to fronteggAuth and replace each with fronteggAuth.isAuthRoutes..

File: packages/vue/src/auth/mapAuthState.ts (hunk 3 above)
Comment (frontegg-david) [reply packages/vue/src/auth/mapAuthState.ts]: removed

File: packages/vue/src/auth/service.ts (hunk 4 above)
Comment (frontegg-david) [reply packages/vue/src/auth/service.ts]: we cannot add types for this one, I fixed the fullPath at line 164, we need to replace it with router proxy to support multiple versions.

File: packages/vue/src/auth/mapAuthState.ts [hunk 8]
```diff
@@ -215,8 +215,8 @@ export const useFrontegg = () => {
   const fronteggStore = useFronteggStore() as EnhancedStore;
 
   const loginWithRedirect = () => {
-    // @ts-ignore
-    if (!fronteggAuth.router?.currentRoute.path.startsWith(authState.routes.hostedLoginRedirectUrl)) {
+    const path = fronteggAuth.getCurrentRoute()
+    if (!path.startsWith(authState.routes.hostedLoginRedirectUrl ?? "/oauth/callback")) {
```
[the remark this answers is not in the document]
Comment (frontegg-david) [reply packages/vue/src/auth/mapAuthState.ts]: fixed

**Review by yuvalotem1 — APPROVED** [review]

**Pull request merged**